src/plans/conjugate_gradient_plan.jl

```diff
@@ -507,3 +508,75 @@ function (u::SteepestDirectionUpdateRule)(
 )
     return 0.0
 end
+
+@doc raw"""
+    ConjugateGradientRestart <: DirectionUpdateRule
+
+An update rule might require a restart, that is one gradient step, if the last two gradients
```

Comment:
```suggestion
An update rule might require a restart, that is using pure gradient as descent direction, if the last two gradients
```
I think this explanation is more clear.

Reply:
By the way, I've also seen restarting when beta gets negative, see here for example: https://en.wikipedia.org/wiki/Nonlinear_conjugate_gradient_method .

Reply:
the PR-CG includes that, but currently it is not a decorator. But we could add that as an option as well.